src/utils/os.h

```diff
@@ -609,6 +617,18 @@ ssize_t read_file(char *path, uint8_t **out);
  */
 int read_file_string(char *path, char **out);
 
+/**
+ * @brief Opens a file for writing and write a
+ * a buffer in nonblocking mode
+ *
+ * @param path[in] The file path
+ * @param fd[out] The file descriptor
```

Comment:
```suggestion
 * @param fd[in, out] The file descriptor.
 * If this is non-`0`, opens the file given by @p path,
 * and sets @p fd to the new file descriptor.
 * Otherwise, @p path is ignored, and @p fd is used.
 * Please close the created @p fd when done with it.
```